packages/calcite-components/src/components/popover/popover.tsx

```diff
@@ -293,7 +293,25 @@ export class Popover
     this.setFilteredPlacements();
     connectLocalized(this);
     connectMessages(this);
-    connectFocusTrap(this);
+    connectFocusTrap(this, {
+      focusTrapEl: this.el,
+      focusTrapOptions: {
+        allowOutsideClick: true,
+        clickOutsideDeactivates: (event: MouseEvent) => {
+          const path = event.composedPath();
+          const isReferenceElementInPath =
+            this.referenceElement instanceof EventTarget && path.includes(this.referenceElement);
```

Comment:
I think this should use `this.effectiveReferenceElement` instead of `this.referenceElement`